pkg/k8s/pipelineRun.go

```diff
@@ -319,7 +332,15 @@ func (r *pipelineRun) updateFinalizers(finalizerList []string) error {
 	return nil
 }
 
-// changeStatusAndUpdateSafely executes `change` and writes the
+func (r *pipelineRun) registerChange(change func() error) error {
```

Comment:
We should find a better name since the change is not only "registered", but also applied.

Reply:
some proposals
- changeInMemory
- changeLocally
- changeAndStoreForRetry

Reply:
Changed to `changeAndStoreForRetry`